contrib/devtools/README.md

```diff
@@ -2,6 +2,47 @@ Contents
 ========
 This directory contains tools for developers working on this repository.
 
+archive-binaries.py
+===============
+
+Script to copy all the binaries (bitcoind, bitcoin-tx, bitcoin-cli,
+qt/bitcoin-qt) to a bin and bin/qt directories (assumed to exist in
+the calling directory), appending the last commit id (and optionally
+an arbitrary string before the commit id passed ar parameter before
+the commit id) to the file name. 
+
+This way one can keep a collection of binaries for many different
+branches for testing (while clearly ditinguishing which is which).
+
+Example usage:
+
+```
+git fetch origin
+git checkout origin/v0.12.0
+make check -j21
+python contrib/devtools/archive-binaries.py v0.12.0
+./bin/bitcoind-v0.12.0-188ca9c
+
+git checkout origin/master
+git clean -fdx -e bin/
```

Comment:
try `make distclean`, this *should* leave the tree clean as it was in git. If not, that's a bug.Of course it will only clean up files it created itself.